property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
```

Comment:
This is the kind of issues that we routinely fix as we discover them. 

Preventing an empty custom type is a matter of adding a 3 lines-long function to `validate` property of the `customType` prompt. Actually, the code is already there, see [property/index.js#L104](https://github.com/strongloop/generator-loopback/blob/f308e4818139192e34734e836928cfeca60b3303/property/index.js#L104).

In my understanding of the current code base (which may be wrong, you are welcome to prove me wrong!), `def.type` will be always defined, therefore there is no handle falsey `def.type` value here. If there are unit-tests that pass invalid `type` or `customType` answer, then we should fix them.